test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -0,0 +1,121 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+
+
+class TestSchedPreemptEnforceResumption(TestFunctional):
+    """
+    Test sched_preempt_enforce_resumption working
+    """
+    def setUp(self):
+        TestFunctional.setUp(self)
+        # Set default resources available on the default mom
+        a = {ATTR_rescavail + '.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        # Create an express queue
+        b = {ATTR_qtype: 'Execution', ATTR_enable: 'True',
+             ATTR_start: 'True', ATTR_p: '151'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, b, "expressq")
+        a = {ATTR_sched_preempt_enforce_resumption: True}
+        self.server.manager(MGR_CMD_SET, SCHED, a)
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+    def test_filler_job(self):
+        """
+        This test confirms that the filler job does not run if it conflicts
+        with running of a suspended job.
+        """
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 80})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 30})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_l + '.walltime': 90})
+        jid3 = self.server.submit(j3)
+        starttime = int(time.time())
```

Comment:
I'd say either move start time before you submit the job or even better to have job id as part of log message. because it may happen that between submit and time the second changes and the log message turns out to be in the previous second.